prototype/southbound/mqtt/mqtt-device-factory/src/main/java/org/eclipse/sensinact/gateway/southbound/mqtt/factory/MqttDeviceFactoryHandler.java

```diff
@@ -0,0 +1,146 @@
+
+    /**
+     * Device mapping service
+     */
+    @Reference
+    private IDeviceMappingHandler mappingHandler;
+
+    /**
+     * Device mapping service configuration
+     */
+    private DeviceMappingConfigurationDTO mappingConfiguration;
+
+    /**
+     * ID of the MQTT handler that we can accept messages from, if defined
+     */
+    private String allowedHandlerId;
+
+    /**
+     * Listener service registration
+     */
+    private ServiceRegistration<IMqttMessageListener> svcReg;
+
+    /**
+     * Component activated
+     */
+    @Activate
+    void activate(final ComponentContext componentContext, final MqttDeviceFactoryConfiguration configuration) {
+        // Check MQTT configuration
+        final String handlerId = configuration.mqtt_handler_id();
+        String[] topics = configuration.mqtt_topics();
+        if ((handlerId == null || handlerId.isBlank()) && (topics == null || topics.length == 0)) {
+            throw new IllegalArgumentException(
+                    "MQTT device factory requires at least a handler ID or a topic to be configured");
+        }
+
+        allowedHandlerId = handlerId;
+
+        if (topics == null || topics.length == 0) {
+            topics = new String[] { "#" };
+        } else if (topics.length == 1 && topics[0].contains(",")) {
+            topics = topics[0].split(",");
+        }
+
+        // Check mapping configuration
+        if (configuration.mapping() == null) {
+            throw new IllegalArgumentException("No mapping configuration given");
+        }
+
+        // Convert the mapping configuration
+        try {
+            mappingConfiguration = new ObjectMapper().readValue(configuration.mapping(),
+                    DeviceMappingConfigurationDTO.class);
+        } catch (JsonProcessingException e) {
+            logger.error("Error parsing MQTT mapping configuration", e);
+            throw new IllegalArgumentException("Invalid mapping configuration", e);
+        }
+
+        // Register the listener service
+        final BundleContext context = componentContext.getBundleContext();
+        final Hashtable<String, Object> properties = new Hashtable<>();
+        if (topics != null) {
+            properties.put(MQTT_TOPICS_FILTERS, topics);
+        }
+        svcReg = context.registerService(IMqttMessageListener.class, this, properties);
+    }
+
+    /**
+     * Component deactivated
+     */
+    @Deactivate
+    void deactivate() {
+        if (svcReg != null) {
+            svcReg.unregister();
+            svcReg = null;
+        }
+    }
+
+    @Override
+    public void onMqttMessage(String handlerId, String topic, IMqttMessage message) {
+        if (allowedHandlerId != null && !allowedHandlerId.equals(handlerId)) {
+            // Ignore message
+            return;
+        }
+
+        // TODO: define the correct session
+        final SensiNactSession session = sessionManager.getDefaultSession("user");
```

Comment:
This doesn't feel quite right. Should we be using push updates and no session? Probably one to discuss in Grenoble as part of the security discussion.

Reply:
This is a relic from the previous Device Factory Handler API, which now uses the `PrototypePush` API. I'm updating the call here.